cg/meta/workflow/rnafusion.py

```diff
@@ -0,0 +1,299 @@
+            return "0.0.0"
+
+    @staticmethod
+    def __build_command_str(options: dict) -> List[str]:
+        formatted_options = []
+        for key, val in options.items():
+            if val:
+                formatted_options.append(str(key))
+                formatted_options.append(str(val))
+        return formatted_options
+
+    def config_case(
+        self,
+        case_id: str,
+        strandedness: str,
+    ) -> None:
+        """Create samplesheet file for RNAFUSION analysis"""
+        self.make_case_folder(case_id)
+        if not (self.get_case_config_path(case_id)).is_file():
+            LOG.info("Samplesheet does not exist, writing samplesheet")
+            self.write_samplesheet(case_id, strandedness)
+        LOG.info("Samplesheet written")
+
+    def run_analysis(
+        self,
+        case_id: str,
+        log: str,
+        work_dir: str,
+        resume: bool,
+        profile: str,
+        with_tower: bool,
+        stub: bool,
+        input: str,
+        outdir: str,
+        genomes_base: str,
+        trim: bool,
+        fusioninspector_filter: bool,
+        all: bool,
+        pizzly: bool,
+        squid: bool,
+        starfusion: bool,
+        fusioncatcher: bool,
+        arriba: bool,
+        dry_run: bool = False,
+    ) -> None:
+        """Execute RNAFUSION run analysis with given options"""
+
+        options = self.__build_command_str(
+            self.get_verified_arguments(
+                case_id=case_id,
+                work_dir=work_dir,
+                resume=resume,
+                profile=profile,
+                with_tower=with_tower,
+                stub=stub,
+                input=input,
+                outdir=outdir,
+                genomes_base=genomes_base,
+                trim=trim,
+                fusioninspector_filter=fusioninspector_filter,
+                all=all,
+                pizzly=pizzly,
+                squid=squid,
+                starfusion=starfusion,
+                fusioncatcher=fusioncatcher,
+                arriba=arriba,
+            )
+        )
+        nextflow_options = self.__build_command_str(
+            self.get_verified_arguments_nextflow(
+                case_id=case_id,
+                log=log,
+            )
+        )
+        command = ["run", self.nfcore_pipeline_path]
+        parameters = (
+            nextflow_options
+            + ["-bg", "-q"]
+            + command
+            + options
+            + [
+                " > "
+                + str(self.get_case_path(case_id))
+                + "/"
+                + case_id
+                + "-stdout.log 2> "
+                + str(self.get_case_path(case_id))
+                + "/"
+                + case_id
+                + "-stdout.err  < /dev/null & "
+            ]
+        )
+        self.process.run_command(parameters=parameters, dry_run=dry_run)
+
+    def get_cases_to_analyze(self) -> List[models.Family]:
```

Comment:
You can disregard the comment, I noticed that there is a base function in the AnalysisAPI, which I think you can use instead of mutating it here. At least try with that first

Reply:
Resolved by removing the function